pkg/demoinfocs/common/equipment.go

```diff
@@ -276,13 +336,8 @@ const weaponPrefix = "weapon_"
 func MapEquipment(eqName string) EquipmentType {
 	eqName = strings.TrimPrefix(eqName, weaponPrefix)
 
-	var wep EquipmentType
-	if strings.Contains(eqName, "knife") || strings.Contains(eqName, "bayonet") {
-		wep = EqKnife
-	} else {
-		// If the eqName isn't known it will be EqUnknown as that is the default value for EquipmentType
-		wep = eqNameToWeapon[eqName]
-	}
+	// If the eqName isn't known it will be EqUnknown as that is the default value for EquipmentType
+	wep := eqNameToWeapon[eqName]
```

Comment:
hmm, this doesn't really scale well for when they add new knifes - we would need to update the parser every time. and worse, our consumers would need to upgrade, otherwise they will get `EqUnknown` for the new knife, which would be really hard to debug.

I suggest that we instead expose the raw `eqName` somewhere upstream in the call chain

Reply:
That is a good point, I will take a look at it!

Reply:
I don't think there is a good way to handle valve adding new knives. However, I think there are some options:

1. Exposing the weapon "code"/itemindex - eg. we give the consumers the possibility to see what code the EqUnkown has.
    - Pros: 
        * This technically means that consumers can still handle new knives even before the parser is updated.
   - Cons:
       * It will still to a degree be hard for consumers to debug, as they only get a weapon code and nothing "human readable". 
       * Parser will still need to be updated to handle new knives with EqTypes and so forth. 

2. Exposing a more raw weapon string. 
    - Pros:
        * Makes it more human readable for consumers to debug.
    - Cons:
        * The parser needs to be updated before consumers can add new knives.
        
After writing this out I find myself gravitating more towards option 1. As the raw weapon string is not something coming from the engine but something we attach from the item index.

Reply:
also why not keeping `strings.Contains(eqName, "knife")` as a fallback when `wep` is unknown?
i think it's better to have `weapon_knife` instead of "Unknown" when we know it's a knife

Reply:
> also why not keeping `strings.Contains(eqName, "knife")` as a fallback when `wep` is unknown? i think it's better to have `weapon_knife` instead of "Unknown" when we know it's a knife

My thought was that the parser defines what a knife "is". Therefore it would be an error in the parser if the EqName did not map to an EqType.

I can add it back, I guess it is nice to have if new knives are added and the eqname/eqtype map is not updated.

Reply:
Actually - I don't think we can add new `EqKnifeXYZ` types anyway as it's not backwards compatible and would lead to confusion (e.g. if the user wants to check if a kill was a knife kill, using `EqKnife` is more straight forward than checking all cases.

So we should expose this as additional data, not as replacement.

So we'll need another way of solving this - e.g. exposing the input string higher up

Reply:
What about the idea of adding a field to the equipment struct with additional detailes about the weapon/equipment? 

Let me know if there is something I can do. :)